filters/builtin/compress.go

```diff
@@ -251,13 +251,15 @@ func canEncodeEntity(r *http.Response, mime []string) bool {
 
 func (c *compress) acceptedEncoding(r *http.Request) string {
 	var encs encodings
-	for _, s := range strings.Split(r.Header.Get("Accept-Encoding"), ",") {
-		sp := strings.Split(s, ";")
-		if len(sp) == 0 {
+
+	for s := range splitSeq(r.Header.Get("Accept-Encoding"), ",") {
+
+		name, spi, isWeighted := strings.Cut(s, ";")
+		if name == "" {
```

Comment:
I think we should trim space before this check: lets add a test case for malformed `Accept-Encoding: foo, ;q=3`